falcon/util/misc.py

```diff
@@ -75,7 +75,9 @@
 
 # PERF(kgriffs): Avoid superfluous namespace lookups
 strptime: Callable[[str, str], datetime.datetime] = datetime.datetime.strptime
-utcnow: Callable[[], datetime.datetime] = datetime.datetime.utcnow
+utcnow: Callable[[], datetime.datetime] = functools.partial(
```

Comment:
Maybe instead using partial we could actually create a function `utcnow`?

Reply:
What is the advantage of creating it? Do you think we could write it in a slightly more performant way?

Reply:
```
def utcnow() -> datetime.datetime:
      return datetime.datetime.now(datetime.timezone.utc)
```
It does exactly the same thing, does not use functools and is one line shorter :)

Reply:
Aha, I see how you reason then :slightly_smiling_face: 

I will, however, stick to `functools`, as I don't see anything wrong with them, and it depends how you look at it regarding the brevity of the two versions. If you disregard formatting/wrapping, the functools variant is actually a oneliner. :smiling_imp:

What is more, the `functools` version seems to be faster:
```python
>>> import timeit
>>> def original(param=None):
...     return param
... 
>>> import functools
>>> import timeit
>>> def original(param=None):
...     return param
... 
>>> def wrapped1():
...     return original('UTC')
... 
>>> wrapped2 = functools.partial(original, 'UTC')
>>> 
>>> timeit.repeat(wrapped1)
[0.07910766499844613, 0.07443887900080881, 0.07431553800051915, 0.074646307999501, 0.07460912899841787]
>>> 
>>> timeit.repeat(wrapped1)
[0.07943654300106573, 0.0741412629995466, 0.07445096799892781, 0.07412566799939668, 0.07406616599837434]
>>> 
>>> timeit.repeat(wrapped2)
[0.050794496999515104, 0.045646627000678563, 0.04568123100034427, 0.045801513999322196, 0.0458722419989499]
>>> 
>>> timeit.repeat(wrapped2)
[0.05152076400008809, 0.046164414001395926, 0.045919030000732164, 0.046146496999426745, 0.046126109000397264]
```

Reply:
usually partial is a bit faster. probably not that important here

Reply:
partial is faster than defining function in pure python? 😯

Reply:
partial is a class implemented in C, so I guess they use a fast path to execute a function and/or resolve the function at construction time vs when it's called